src/reporting/Market.sol

```diff
@@ -0,0 +1,653 @@
+// Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/reporting/Branch.sol';
+import 'ROOT/reporting/ReportingToken.sol';
+import 'ROOT/reporting/ReputationToken.sol';
+import 'ROOT/reporting/DisputeBondToken.sol';
+import 'ROOT/reporting/Interfaces.sol';
+import 'ROOT/trading/Cash.sol';
+import 'ROOT/libraries/DelegationTarget.sol';
+import 'ROOT/extensions/MarketFeeCalculator.sol';
+import 'ROOT/factories/MapFactory.sol';
+import 'ROOT/factories/ShareTokenFactory.sol';
+import 'ROOT/factories/ReportingTokenFactory.sol';
+import 'ROOT/factories/DisputeBondTokenFactory.sol';
+import 'ROOT/libraries/Typed.sol';
+import 'ROOT/libraries/Initializable.sol';
+import 'ROOT/libraries/token/ERC20Basic.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+import 'ROOT/libraries/math/SafeMathInt256.sol';
+
+
+contract Market is DelegationTarget, Typed, Initializable {
+    using SafeMathUint256 for uint256;
+    using SafeMathInt256 for int256;
+
+    enum ReportingState {
+        AUTOMATED,
+        LIMITED,
+        ALL,
+        FORK,
+        FINALIZED
+    }
+
+    ReportingState private reportingState = ReportingState.AUTOMATED;
+
+    // CONSIDER: change the payoutNumerator/payoutDenominator to use fixed point numbers instead of integers; PRO: some people find fixed point decimal values easier to grok; CON: rounding errors can occur and it is easier to screw up the math if you don't handle fixed point values correctly
+    uint256 private payoutDenominator;
+    uint256 private feePerEthInAttoeth;
+
+    // CONSIDER: we really don't need these
+    int256 private maxDisplayPrice;
+    int256 private minDisplayPrice;
+
+    // CONSIDER: figure out approprate values for these
+    uint256 private constant AUTOMATED_REPORTER_DISPUTE_BOND_AMOUNT = 11 * 10**20;
+    uint256 private constant LIMITED_REPORTERS_DISPUTE_BOND_AMOUNT = 11 * 10**21;
+    uint256 private constant ALL_REPORTERS_DISPUTE_BOND_AMOUNT = 11 * 10**22;
+
+    uint256 private constant MIN_NUM_OUTCOMES = 2;
+    uint256 private constant MIN_PAYOUT_DENOMINATOR = 2;
+    uint256 private constant MAX_FEE_PER_ETH_IN_ATTOETH = 5 * 10 ** 17;
+    int256 private constant DISPLAY_PRICE_MIN = -(2 ** 254);
+    int256 private constant DISPLAY_PRICE_MAX = 2 ** 254;
+    uint256 private constant SET_COST_MULTIPLIER_MAX = 2 ** 254;
+    uint256 private constant APPROVAL_AMOUNT = 2 ** 254;
+    uint256 private constant AUTOMATED_REPORTING_DURATION_SECONDS = 3 days;
+    uint256 private constant AUTOMATED_REPORTING_DISPUTE_DURATION_SECONDS = 3 days;
+    address private constant NULL_ADDRESS = address(0);
+
+    ReportingWindow private reportingWindow;
+    uint256 private endTime;
+    uint8 private numOutcomes;
+    uint256 private marketCreationBlock;
+    bytes32 private topic;
+    address private automatedReporterAddress;
+    mapping(bytes32 => ReportingToken) private reportingTokens;
+    Cash private denominationToken;
```

Comment:
Suggest renaming this `cash` since that is now what it is (same comment all other places this variable is used).